lib/vagrant-libvirt/action/forward_ports.rb

```diff
@@ -109,7 +110,7 @@ def redirect_port(machine, host_ip, host_port, guest_ip, guest_port,
           options += " -o ProxyCommand=\"#{ssh_info[:proxy_command]}\"" if machine.provider_config.connect_via_ssh
 
           # TODO: instead of this, try and lock and get the stdin from spawn...
-          ssh_cmd = 'exec '
```

Comment:
This was originally added to ensure a `vagrant destroy` quickly followed by `vagrant up` would work as expected as previously the ssh process was not getting removed by a destroy. However your commit suggests that this shouldn't have worked as it should still have been getting the wrong PID back?

Reply:
Exactly.  Moreover, the SSH process used might even be from one with multiplexing enabled (`ControlMaster=auto` in ssh config) so there might not even be a new process at all (e.g. from a previous destroy/up cycle, or from other/externally managed SSH processes.)